api/api/utils/check_dead_links/__init__.py

```diff
@@ -43,22 +43,34 @@ def _get_expiry(status, default):
 _timeout = aiohttp.ClientTimeout(total=2)
 
 
-async def _head(url: str) -> tuple[str, int]:
+async def _head(url: str, session: aiohttp.ClientSession) -> tuple[str, int]:
+    start_time = time.time()
```

Comment:
The [`perf_counter` function](https://docs.python.org/3/library/time.html#time.perf_counter) is better suited here. It's more precise to measure a short duration. It will need to be changed in the end_time too.
```suggestion
    start_time = time.perf_counter()
```

Reply:
TIL thanks! Pushed the change in the next commit.